irispy/sji.py

```diff
@@ -2,880 +2,550 @@
-        return (
-            """SunPy {dtype!s}
----------
-Observatory:\t {obs}
-Instrument:\t {inst}
-Detector:\t {det}
-Measurement:\t {meas}
-Wavelength:\t {wave}
-Obs. Start:\t {date_start:{tmf}}
-Obs. End:\t {date_end:{tmf}}
-Num. of Frames:\t {frame_num}
-IRIS Obs. id:\t {obs_id}
-IRIS Obs. Description:\t {obs_desc}
-Dimensions:\t {dim}
-Scale:\t\t {scale}
-""".format(dtype=self.__class__.__name__,
-                obs=self.observatory, inst=self.instrument, det=self.detector,
-                meas=self.measurement, wave=self.wavelength, date_start=self.date[0],
-                date_end=self.date[-1], frame_num=len(self),
-                dim=u.Quantity(self.dimensions),
-                scale=u.Quantity(self.scale), obs_id=self.iris_obs_id, obs_desc=self.iris_obs_description,
-                tmf=TIME_FORMAT) + self.data.__repr__())
-
-    # Sorting methods
-    @classmethod
-    def _sort_by_date(cls):
-        return lambda m: m.date  # maps.sort(key=attrgetter('date'))
-
-    def _derotate(self):
-        """Derotates the layers in the MapCube"""
-        pass
-
-    @property
-    def iris_obs_id(self):
-        """IRIS Observation ID"""
-        return self._meta[self.ref_index].get('OBSID', "")
-
-    @property
-    def iris_obs_description(self):
-        """IRIS Observation Description"""
-        return self._meta[self.ref_index].get('OBS_DESC', "")
-
-    @property
-    def instrument(self):
-        """Instrument name"""
-        return self._meta[self.ref_index].get('instrume', "").replace("_", " ")
-
-    @property
-    def measurement(self):
-        """Measurement name, defaults to the wavelength of image"""
-        return u.Quantity(self._meta[self.ref_index].get('wavelnth', 0), self._meta[self.ref_index].get('waveunit', ""))
-
-    @property
-    def wavelength(self):
-        """wavelength of the observation"""
-        return u.Quantity(self._meta[self.ref_index].get('wavelnth', 0), self._meta[self.ref_index].get('waveunit', ""))
-
-    @property
-    def observatory(self):
-        """Observatory or Telescope name"""
-        return self._meta[self.ref_index].get('obsrvtry', self._meta[self.ref_index].get('telescop', "")).replace("_", " ")
-
-    @property
-    def detector(self):
-        """Detector name"""
-        return self._meta[self.ref_index].get('detector', "")
 
-    @property
-    def dimensions(self):
+    def __init__(self, data, wcs, uncertainty=None, unit=None, meta=None,
+                 mask=None, extra_coords=None, copy=False, missing_axis=None,
+                 scaled=None):
         """
-        The dimensions of the array (x axis first, y axis second).
+        Initialization of Slit Jaw Imager
         """
-        return self._meta[0].get('NAXIS1'), self._meta[0].get('NAXIS2')
+        warnings.warn("This class is still in early stages of development. API not stable.")
+        # Set whether SJI data is scaled or not.
+        self.scaled = scaled
+        # Set the dust mask for the data
```

Comment:
Replace by something like "Set the dust_masked variable to False because the dust mask not applied. This variable will be modified to True if the dust mask is applied."